### PR TITLE
Set password

### PR DESCRIPTION
**Pull request by jonallured**
This PR is for this ticket: https://artsyproduct.atlassian.net/browse/SELL-92, which is some fallout from work done to improve security on waves. What I'm trying to do here is modify the forgot password flow so that it's suitable for fair partners that have not yet set their password. The current changes in this PR all pretty much all cosmetic, but you gotta start somewhere!

What I've done is add some additional info from the url so that I can conditionally show and hide content on the forgot password modal and reset password page. I've stopped here to ensure I'm on the right track. The next steps would be:

* take a look at the page after /forgot to ensure the copy lines up
* change the email sent to include the redirect_to param
* take a look at the page after /reset_password to ensure the copy lines up

Basically, I need to make the flow actually work. 😝

### CODE REVIEW

File: src/desktop/apps/auth/routes.coffee [hunk 1]
```diff
@@ -8,9 +8,12 @@ sanitizeRedirect = require '@artsy/passport/sanitize-redirect'
 @resetPassword = (req, res) ->
   if req.query.reset_password_token?
     req.session.reset_password_token = req.query.reset_password_token
+    req.session.set_password = req.query.set_password
+    req.session.redirect_to = req.query.redirect_to
     res.redirect '/reset_password'
   else
-    res.render 'reset_password', reset_password_token: req.session.reset_password_token
+    res.locals.sd.REDIRECT_TO = req.session.redirect_to
```
Comment (mzikherman) [inline src/desktop/apps/auth/routes.coffee]: Maybe a more specific name for this? Something like `RESET_PASWORD_REDIRECT_TO` or something.

**Comment by jonallured** [comment]
@mzikherman ok, this is passing CI and I think it's mostly there. When we talked the other day, you mentioned some tests that might be good to add, but I didn't take good enough notes - can you point me in the right direction? Or if it's not a big deal, I'm happy to move forward - whatever you think.

**Comment by mzikherman** [comment]
Looks good!

In terms of tests:
- maybe testing that the 'mode' and 'redirect to' is being properly passed to the 'send reset instructions' endpoint (which will use that to generate the correct link). You can check out the existing tests which assert the correct API call w/ params is made: https://github.com/artsy/force/blob/ed638bd67ce0fe52eadac0fdf26f092b4a49855f/src/desktop/test/models/logged_out_user.coffee#L80

**Comment by jonallured** [comment]
Ok, I think this is all set @mzikherman! I added some tests with https://github.com/artsy/force/pull/2321/commits/ab3dbf53d2195914a8d413eaf5d94f2bc2ac579a that I think look sharp, but let me know if you think there's anything else to do here!

**Comment by mzikherman** [comment]
🎉 LGTM!
